sqlmesh/dbt/adapter.py

```diff
@@ -295,41 +302,42 @@ def execute(
         return AdapterResponse("Success"), empty_table()
 
     def resolve_schema(self, relation: BaseRelation) -> t.Optional[str]:
-        schema = self._map_table_name(relation.database, relation.schema, relation.identifier).db
-        if not schema:
-            return None
-        return schema
+        schema = self._map_table_name(self._normalize(self._relation_to_table(relation))).db
+        return schema if schema else None
 
     def resolve_identifier(self, relation: BaseRelation) -> t.Optional[str]:
-        identifier = self._map_table_name(
-            relation.database, relation.schema, relation.identifier
-        ).name
-        if not identifier:
-            return None
-        return identifier
+        identifier = self._map_table_name(self._normalize(self._relation_to_table(relation))).name
+        return identifier if identifier else None
 
-    def _map_table_name(
-        self, database: t.Optional[str], schema: t.Optional[str], identifier: t.Optional[str]
-    ) -> exp.Table:
-        name = ".".join(p for p in (database, schema, identifier) if p is not None)
+    def _map_table_name(self, table: exp.Table) -> exp.Table:
+        name = exp.table_name(table, dialect=self.engine_adapter.dialect)
         if name not in self.table_mapping:
-            return exp.to_table(name, dialect=self.engine_adapter.dialect)
+            return table
 
         physical_table_name = self.table_mapping[name]
         logger.debug("Resolved ref '%s' to snapshot table '%s'", name, physical_table_name)
 
         return exp.to_table(physical_table_name, dialect=self.engine_adapter.dialect)
 
     def _relation_to_table(self, relation: BaseRelation) -> exp.Table:
-        assert relation.identifier is not None
-        return exp.table_(relation.identifier, db=relation.schema, catalog=relation.database)
+        table = exp.to_table(relation.render(), dialect=self.engine_adapter.dialect)
+        return exp.to_table(relation.render(), dialect=self.engine_adapter.dialect)
+
+    def _table_to_relation(self, table: exp.Table) -> BaseRelation:
+        return self.relation_type.create(
+            database=table.catalog or None,
```

Comment:
why are you calling table.catalog? you lose quotes if you do that

Reply:
mimics what dbt does